plugins/woocommerce/templates/loop/orderby.php

```diff
@@ -12,16 +12,21 @@
  *
  * @see         https://woocommerce.com/document/template-structure/
  * @package     WooCommerce\Templates
- * @version     3.6.0
+ * @version     9.6.0
  */
 
 if ( ! defined( 'ABSPATH' ) ) {
 	exit;
 }
 
+$id_suffix = wp_unique_id();
+
 ?>
 <form class="woocommerce-ordering" method="get">
-	<select name="orderby" class="orderby" aria-label="<?php esc_attr_e( 'Shop order', 'woocommerce' ); ?>">
+	<?php if ( $use_label ) : ?>
+		<label for="woocommerce-orderby-<?php echo esc_attr( $id_suffix ); ?>"><?php echo esc_html__( 'Sort by', 'woocommerce' ); ?></label>
+	<?php endif; ?>
+	<select name="orderby" class="orderby" <?php echo $use_label ? "id='woocommerce-orderby-" . esc_attr( $id_suffix ) . "'" : "aria-label='" . esc_attr__( 'Shop order', 'woocommerce' ) . "'"; ?>>
```

Comment:
Nitpick: This code is looking a bit harder to read. It would be great if we can make it more readable. For examples, instead of using `echo`, we can consider using  `esc_attr_e()`, `esc_html_e()` etc. Here is what I have in my mind. I know we were already using `echo` in this file, but this could be a good opportunity to make it more readable.

Note: I haven't tested below code. It's just a suggestion.

```php
<form class="woocommerce-ordering" method="get">
	<?php if ( $use_label ) : ?>
		<label for="<?php esc_attr_e( "woocommerce-orderby-{$id_suffix}" ); ?>">
			<?php esc_html_e( 'Sort by', 'woocommerce' ); ?>
		</label>
	<?php endif; ?>

	<select
		name="orderby"
		class="orderby"
		<?php if ( $use_label ) : ?>
			id="<?php esc_attr_e( "woocommerce-orderby-{$id_suffix}" ); ?>"
		<?php else : ?>
			aria-label="<?php esc_attr_e( 'Shop order', 'woocommerce' ); ?>"
		<?php endif; ?>
	>
		<?php foreach ( $catalog_orderby_options as $id => $name ) : ?>
			<option value="<?php esc_attr_e( $id ); ?>" <?php selected( $orderby, $id ); ?>>
				<?php esc_html_e( $name ); ?>
			</option>
		<?php endforeach; ?>
	</select>

	<input type="hidden" name="paged" value="1" />
	<?php wc_query_string_form_fields( null, array( 'orderby', 'submit', 'paged', 'product-page' ) ); ?>
</form>
```